Code/Geometry/point.h

```diff
@@ -70,24 +70,36 @@ class RDKIT_RDGEOMETRYLIB_EXPORT Point3D : public Point {
   inline unsigned int dimension() const override { return 3; }
 
   inline double operator[](unsigned int i) const override {
-    PRECONDITION(i < 3, "Invalid index on Point3D");
-    if (i == 0) {
-      return x;
-    } else if (i == 1) {
-      return y;
-    } else {
-      return z;
+    switch (i) {
```

Comment:
Nice! this is a better way of handling this that what we had before.

I also think that raising a `ValueError` instead of a `PRECONDITION` failure makes sense. However we should document it and, to be consistent, `PointND` should do the same thing.
So:
1. Please update `operator[]` in PointND to check for bad values of `i` and throw a `ValueError`
2. Add a note to the `Backwards Incompatible Changes` section of `ReleaseNotes.md`

Reply:
why does this change make it no longer backward compatible? (just so I know in future changes)

Reply:
> why does this change make it no longer backward compatible? (just so I know in future changes)

Since the type of exception that would be thrown is different, people may have to update their code if they have logic to catch specific exception types.

This isn't super important, but it's still a good thing to mention.

Reply:
> * Please update `operator[]` in PointND to check for bad values of `i` and throw a `ValueError`

Please ignore this suggestion. The bounds check is covered by the underlying `Vector` object